components/vision.py

```diff
@@ -177,38 +194,43 @@ def desired_turret_rotation(self) -> Rotation2d:
         return self.relative_bearing_to_best_cluster()
 
     def turret_to_servo(self, turret: Rotation2d) -> Rotation2d:
-        return turret - (self.servo_offset - self.encoder_offset)
+        return turret - (self.servo_offsets.neutral - self.encoder_offset)
 
     @property
     def turret_rotation(self) -> Rotation2d:
         return self.raw_encoder_rotation() - self.encoder_offset
 
     def robot_to_camera(self, timestamp: float) -> Transform3d:
-        robot_to_turret_rotation = self.robot_to_turret.rotation()
         turret_rotation = self.turret_rotation_buffer.sample(timestamp)
         if turret_rotation is None:
             return self.robot_to_turret
-        return Transform3d(
-            self.robot_to_turret.translation(),
-            Rotation3d(
-                robot_to_turret_rotation.x,
-                robot_to_turret_rotation.y,
-                robot_to_turret_rotation.z + turret_rotation.radians(),
-            ),
+        r2t = self.robot_to_turret.translation()
+        t2c = self.turret_to_camera.translation().rotateBy(Rotation3d(turret_rotation))
+        trans = Translation3d(r2t.x + t2c.x, r2t.y + t2c.y, r2t.z + t2c.z)
```

Comment:
```suggestion
        trans = r2t + t2c
```

Reply:
I tried that, and apparently you can't add two `Translation3d`s??

Reply:
Weird, works fine in a REPL for me.

Reply:
Okay, now it works for me. Weird.